Dockerfile

```diff
@@ -1,5 +1,6 @@
 #############      builder       #############
-FROM eu.gcr.io/gardener-project/3rd/golang:1.15.5 AS builder
+ARG GCR_PULL_URL
```

Comment:
/invite @AndreasBurger 

I guess the `.ci/pipeline_definitions` file must be enhanced as well so that the CI/CD system will use the GCR_PULL_URL like before as well.

Reply:
I'd suggest to use a default value directly in the Dockerfile i.e., `ARG GCR_PULL_URL=eu.gcr.io/gardener-project/3rd`

Reply:
@timuthy and pass blank/empty string while 'make'ing for PPC and other ports going forward, right?

Reply:
Yes, although you need to twist this logic in the Makefile again, so that we reach the following:

`make docker-images` --> `... @docker build --build-arg EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) -t $(APISERVER_IMAGE_REPOSITORY):$(EFFECTIVE_VERSION)          -t $(APISERVER_IMAGE_REPOSITORY):latest          -f Dockerfile --target apiserver . ...`

--> uses default value provided in Dockerfile (no build arg)

and 

`make docker-images GCR_PULL_URL=''` --> `... @docker build --build-arg EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) --build-arg GCR_PULL_URL='' -t $(APISERVER_IMAGE_REPOSITORY):$(EFFECTIVE_VERSION)          -t $(APISERVER_IMAGE_REPOSITORY):latest          -f Dockerfile --target apiserver . ...`

--> uses value provided from outside

Feel free to choose a more elegant way but at the moment this is the only option I can think of.